pkg/utils/utils_linux.go

```diff
@@ -91,21 +98,30 @@ func determineEndian() binary.ByteOrder {
 	return endian
 }
 
-// Get all the veth interfaces.
-// Similar to ip link show type veth
-func GetInterface(name string, interfaceType string) (netlink.Link, error) {
-	links, err := showLink()
+// GetOutgoingInterface gets the outgoing interface by executing an equivalent to `ip route show default`
+func GetOutgoingInterface() (netlink.Link, error) {
+	routes, err := routeList(nil, netlink.FAMILY_V4)
```

Comment:
In what cases would err not be nil? 

Does this code change support dual stack deployments?